cloudinit/cmd/devel/shell_script_per_boot.py

```diff
@@ -0,0 +1,25 @@
+# part-handler
+# vi: syntax=python ts=4
+
+import os
+from cloudinit import log
+from cloudinit import util
+
+LOG = log.getLogger(__name__)
+
+
+def list_types():
+    # return a list of mime-types that are handled by this module
+    LOG.debug("in shell_script-per-boot.list_types() ...")
+    return(["text/x-shellscript-per-boot"])
+
+
+def handle_part(data, ctype, script_path, payload):
+    if script_path is not None:
+        LOG.info("in shell_script-per-boot.handle_part() ...")
+        LOG.debug("script_path=%s", script_path)
+        (folder, filename) = os.path.split(script_path)
+        LOG.debug("folder=%s filename=%s", folder, filename)
+        path = f"/var/lib/cloud/scripts/per-boot/{filename}"
```

Comment:
Thanks. I've made this change. Interestingly, the shell_script.py handler actually doesn't do what you described. It does some hocus pocus in the constructor to extract that scripts path, and assign it to a variable, which it then uses later in handle_part. I am not sure which is better -- yours is certainly simpler -- and maybe it'd be better to unify shell_script.py and my new handlers.

Reply:
Correction: I could not get this change to build, because of  a circular reference issue: cloudinit.cmd.devel:__init.py__ imports stages.py, which imports all the handlers including mine, so my handlers can't import cloud.cmd.devel to get that utility method. I ended up copy/pasting the shell_script.py approach.

The method you pointed out seems like a useful method though ... would it make sense to put that functionality up somewhere higher, like in cloudinit.util?